### DISCUSSION TITLE
Plans about Azure DevOps?

### DISCUSSION THREAD

**Discussion by MikeChongCan** [danger/danger-js#1130] [Ideas]
Hi, I found #925 and #1016 haven't been updated for a while and would like to know the plan.

There's an official Azure DevOps client here: https://www.npmjs.com/package/azure-devops-node-api
According to the design of `danger`, can we include this one as a dependency.

**Comment by orta** [comment]
Sure, yeah that dependency seems to make sense for the communication layer 👍🏻

**Reply by MikeChongCan** [reply]
Thanks for your quick reply! Will take a look at ADO support in my spare time.

**Reply by sgtwilko** [reply]
@imWildCat  any new on this?

**Reply by MikeChongCan** [reply]
@sgtwilko unfortunately, no

**Comment by jamiehaywood** [comment]
I'd be keen to look into this

**Comment by krishnasaga** [comment]
I managed to write a function to use Azure DevOps REST API to get the diff between two branches

```javascript

export const getDiff = async ({
  repository,
  baseVersion,
  targetVersion,
  authKey,
}) => {
  const result = await fetch(
    `${AZURE_ORGANIZATION_URL}/_apis/git/repositories/${repository}/diffs/commits?baseVersion=${baseVersion}&targetVersion=${targetVersion}&api-version=7.1-preview.1`,
    {
      method: 'GET',
      headers: {
        Authorization: `Basic ${authKey} `,
      },
    }
  );

  const json = await result.json();

  return json;
};


```

**Reply by krishnasaga** [reply]
I found something strange with this API yesterday I'm getting complete diff, but now I'm getting empty array in `changes: []` 
Do anyone have any idea about this behavior ?

**Comment by gustavolaux** [comment]
any updates on it? i may help in some free time

**Reply by krishnasaga** [reply]
Hi @gustavolaux 

From my side what I can say is, The above issue which I have mentioned is resolved. I would like to integrate this as one of the Provider of DangerJS.

**Reply by krishnasaga** [reply]
@imWildCat @orta @sgtwilko @gustavolaux. I would like to start the work by understanding the provider interface of DangerJs
Do any one here understand how it works ? I'm very keen to contribute. we can accomplish this. 

Or can also help to resolve this issue in this PR  this https://github.com/danger/danger-js/pull/1016  not not sure if it is based on API

**Reply by MikeChongCan** [reply]
@krishnasaga thanks! Please go ahead

**Comment by nvonbenken** [comment]
Any progress on this?
